### PR TITLE
Mapsplice2: Add new package

### PR DESCRIPTION
**Pull request by kuramoto-fj**
Add new package : Mapsplice2
MapSplice is a software for mapping RNA-seq data to reference genome for splice junction discovery that depends only on reference genome, and not on any further annotations.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mapsplice2/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    homepage = "http://www.netlab.uky.edu/p/bioinfo/MapSplice2"
+    url = "http://protocols.netlab.uky.edu/~zeng/MapSplice-v2.2.1.zip"
+
+    version(
+        "2.2.1",
+        sha256="4f3c1cb49ba0abcfc952de5946ee0b56db28c51f4f4d4f5abca66b4461ca7d05",
+    )
+
+    patch("Makefile.patch")
+    patch("mapsplice_ebwt.patch")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mapsplice2/package.py:24]: Have these patches been contributed upstream?

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/mapsplice2/package.py (hunk 1 above)
Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/mapsplice2/package.py:24]: I haven't upstreamed it yet.
However, the last update of mapsplice2 is June 2016, should I upstream it?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/mapsplice2/package.py:24]: It's always worth a shot. If they no longer maintain it, they can just ignore you. But hopefully they'll use it.

Comment (kuramoto-fj) [reply var/spack/repos/builtin/packages/mapsplice2/package.py:24]: Thank you for your reply.
I agree with your opinion. So I'll try upstream patch files.

**Pull request merged**